src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,123 @@
-/* eslint-disable max-len */
-import React from 'react';
+/* eslint-disable no-console */
+/* eslint-disable no-nested-ternary */
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { useDispatch } from 'react-redux';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+
+import { Todo } from '../../types/Todo';
+import { useAppSelector } from '../../app/hooks';
+import { getTodos } from '../../api';
+import { Loader } from '../Loader';
+
+export const TodoList = (
+) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    getTodos().then((todosFromServer) => setTodos(todosFromServer));
+  }, []);
+
+  const filter = useAppSelector((state) => state.filter);
+  const query = useAppSelector((state) => state.filter.query);
+  const currentTodo = useAppSelector((state) => state.currentTodo);
+  const filteredTodos = todos.filter((todo) => {
+    const isMatchingStatus
+      = (filter.status === 'active' && !todo.completed)
+      || (filter.status === 'completed' && todo.completed)
+      || filter.status === 'all';
+
+    if (isMatchingStatus && query.trim() !== '') {
```

Comment:
```suggestion
    if (isMatchingStatus && query.trim()) {
```
empty string is a [falsy ](https://www.freecodecamp.org/news/falsy-values-in-javascript/)value